frontend/internal-packages/agent/src/chat/workflow/nodes/generateUsecaseNode.ts

```diff
@@ -1,13 +1,120 @@
+import { QAGenerateUsecaseAgent } from '../../../langchain/agents'
+import type { Usecase } from '../../../langchain/agents/qaGenerateUsecaseAgent/agent'
+import type { BasePromptVariables } from '../../../langchain/utils/types'
+import { convertSchemaToText } from '../../../utils/convertSchemaToText'
 import type { WorkflowState } from '../types'
 
+const NODE_NAME = 'generateUsecaseNode'
+
+/**
+ * Log usecase generation results for debugging/monitoring purposes
+ * TODO: Remove this function once the feature is stable and monitoring is no longer needed
+ */
+const logUsecaseResults = (
+  logger: WorkflowState['logger'],
+  usecases: Usecase[],
+): void => {
+  logger.log(`[${NODE_NAME}] Generated ${usecases.length} use cases`)
+
+  usecases.forEach((usecase, index) => {
+    logger.log(
+      `[${NODE_NAME}] Usecase ${index + 1} (${usecase.requirementType}): ${JSON.stringify(usecase)}`,
+    )
+  })
+}
+
+/**
+ * Format analyzed requirements into a structured text for AI processing
+ */
+function formatAnalyzedRequirements(
+  analyzedRequirements: NonNullable<WorkflowState['analyzedRequirements']>,
+  userInput: string,
+): string {
+  return `
+Business Requirement: ${analyzedRequirements.businessRequirement}
+
+Functional Requirements:
+${Object.entries(analyzedRequirements.functionalRequirements)
+  .map(
+    ([category, requirements]) =>
+      `${category}:\n${requirements.map((req) => `- ${req}`).join('\n')}`,
+  )
+  .join('\n\n')}
+
+Non-Functional Requirements:
+${Object.entries(analyzedRequirements.nonFunctionalRequirements)
+  .map(
+    ([category, requirements]) =>
+      `${category}:\n${requirements.map((req) => `- ${req}`).join('\n')}`,
+  )
+  .join('\n\n')}
+
+Original User Input: ${userInput}
+`
+}
+
 /**
  * Generate Usecase Node - QA Agent creates use cases
- * Performed by qaAgent
+ * Performed by qaGenerateUsecaseAgent
  */
 export async function generateUsecaseNode(
   state: WorkflowState,
 ): Promise<WorkflowState> {
-  return {
-    ...state,
+  state.logger.log(`[${NODE_NAME}] Started`)
+
+  // Check if we have analyzed requirements
+  if (!state.analyzedRequirements) {
+    const errorMessage =
+      'No analyzed requirements found. Cannot generate use cases.'
+    state.logger.error(`[${NODE_NAME}] ${errorMessage}`)
+    return {
+      ...state,
+      error: errorMessage,
+    }
+  }
+
+  const qaAgent = new QAGenerateUsecaseAgent()
+  const schemaText = convertSchemaToText(state.schemaData)
+
+  // Create a user message that includes the analyzed requirements
+  const requirementsText = formatAnalyzedRequirements(
+    state.analyzedRequirements,
+    state.userInput,
+  )
+
+  const promptVariables: BasePromptVariables = {
+    schema_text: schemaText,
```

Comment:
me too, I didn't think it made sense for all agents to use BasePromptVariables. I think they should be separated from each other.

Reply:
I will try to separate BasePromptVariables in another pull request! Thank you!